### ISSUE TITLE
Broken AI in Exodus (sphere) level

### ISSUE THREAD

**Issue by ccomrade** [crymp-net/client-server#179]
Some AI characters do weird things in Exodus level. For example, Prophet has problems moving. It seems it's impossible to finish the level. No issue in v21. Only current dev builds are affected.

**Comment by Choum28** [comment]
For information about the V21 (main and dev one), the gunner of the first vehicle do not fire, which is also not normal.

**Comment by ccomrade** [comment]
The issue was introduced in https://github.com/crymp-net/client-server/commit/7e42937c245eb1726d13649fdbcd8a1a5b9c24bd by the CryAISystem patch. I'll investigate and prepare a fix.

@griseraner FYI

**Comment by ccomrade** [comment]
The memory patch actually did the opposite and disabled certain AI stuff in both singleplayer and multiplayer. It resulted in all sorts of weird issues in singleplayer. Anyway, the fix is ready and will be available in the dev version once this is closed.

@Choum28 Thanks for reporting the issue!

**Comment by Choum28** [comment]
The new dev version works good for most part (prophet, solider shooting from vehichle etc...)

Helena Rosenthal is suppposed to move into the ADAV for evacuation, she seems block, animation to move seems to begin but loop, she event dissapears during few frame, then reappears at the same place,

After some time, the game will teleport her inside ADAV, preventing mission to be blocked.

In normal behavior, she move herself ([https://youtu.be/3s3OWbIuQR0) see](https://youtu.be/3s3OWbIuQR0?t=458)

**Comment by ccomrade** [comment]
I've just tested it and she walks there on her own. Did you load a saved game before it? That can cause strange issues sometimes.

**Comment by Choum28** [comment]
Yes I load a saved game from previous launcher version.
When I tried to "reload level" from this save, and I have no HUD, Keyboard escape button not working etc...

**Comment by ccomrade** [comment]
If the saved game is from the previous version, it's likely to be corrupted. The issue was that certain hidden AI objects weren't spawned at all, so the game state probably isn't serialized properly in the saved game.

About missing HUD, that sounds like a big issue. How exactly does it happen? Is it the quick load thing?

**Comment by Choum28** [comment]
For the "rossental bug", I tried from the main menu with "map sphere" command. I made a video of the problem.
I have to use one of the checkpoint save (so created with the new version) as I have die one first try to record the problem.

https://www.youtube.com/watch?v=KLdpoGjNNNI


For the Hud problem when restarting level

Start a new game or use any "map XXX", when you have control of your character, press escape -> restart level.
Now you will not be able to skip the intro with space (if new campaign), the escape button will not bring the menu, no hud, if you  wil never have the green message to restart from previous save.
